File: scala/src/main/scala/bron_kerbosch2_gpx.scala
import bron_kerbosch_pivot.PivotChoice.{MaxDegree, MaxDegreeLocalX}
import bron_kerbosch_pivot.visit

import scala.collection.{immutable, mutable}

class bron_kerbosch2_gpx[Vertex] extends bron_kerbosch_algorithm[Vertex] {
  def explore(
      graph: UndirectedGraph[Vertex],
      reporter: immutable.Iterable[Vertex] => Unit
  ): Unit = {
    val candidates = graph.connected_vertices().to(mutable.Set)
    if (candidates.nonEmpty) {
      visit(
        graph,
        reporter,
        MaxDegree,
        MaxDegreeLocalX,
        candidates,
        mutable.Set(),
        immutable.List()
      )
    }
  }
}

File: kotlin/src/main/be.steinsomers.bron_kerbosch/BronKerbosch2gpx.kt
package be.steinsomers.bron_kerbosch

class BronKerbosch2gpx : BronKerboschAlgorithm {
    override val name: String = "Ver2½-GPX"
    override val hasRaceCondition: Boolean = false

    override fun explore(graph: UndirectedGraph, cliqueConsumer: CliqueConsumer) {
        BronKerboschPivot.explore(graph, cliqueConsumer, PivotChoice.MaxDegreeLocalX)
    }
}
